google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineRuntimeInstance.java

```diff
@@ -84,18 +83,42 @@ public void deploy(
                 createdDeployments.put(task.getProject(), deployRunner);
               }
 
-              ProgressManager.getInstance()
-                  .run(
-                      new Task.Backgroundable(
-                          task.getProject(),
-                          GctBundle.message("appengine.deployment.status.deploying"),
-                          true /* canBeCancelled */,
-                          null /* backgroundOption */) {
-                        @Override
-                        public void run(@NotNull ProgressIndicator indicator) {
-                          ApplicationManager.getApplication().invokeLater(deployRunner);
-                        }
-                      });
+              CloudSdkLogger cloudSdkLogger =
```

Comment:
wdyt about un-inlining this to make this method cleaner?

Reply:
just to confirm too; CloudSdkLogger is used to add _Cloud SDK specific_ log messaging to whatever task is being blocked by its installation (e.g. deploy or local-run). This is why you need to have the caller specify this callback (since its specific messaging related to the task, and a specific way to indicate errors. Does that sound correct?

I feel like maybe CloudSdkLogger still isn't the perfect name since it also handles failing the task (e.g. `callback.errorOccured`.

Reply:
Absolutely correct, and yes, I agree about name not being perfect. We might need to make it longer but easier to understand what it does. `CloudSdkFeedbackHandler`?

Reply:
how about `CloudSdkStatusHandler`?

Reply:
Cool name, let's use that!